src/ngx_lua_resty_lmdb_module.c

```diff
@@ -339,6 +415,113 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
 }
 
 
+static ngx_int_t
+ngx_lua_resty_lmdb_close_file(ngx_cycle_t *cycle,
+                              ngx_lua_resty_lmdb_conf_t *lcf)
+{
+    mdb_txn_abort(lcf->ro_txn);
+    mdb_env_close(lcf->env);
+
+    lcf->env = NULL;
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_lua_resty_lmdb_verify_file_status(ngx_cycle_t *cycle,
+                                      ngx_lua_resty_lmdb_conf_t *lcf)
+{
+    ngx_core_conf_t  *ccf;
+    ngx_file_info_t   fi;
+
+    u_char            name_buf[NGX_MAX_PATH];
+    ngx_str_t        *name;
+    ngx_str_t         names[] = { ngx_string("/"),
+                                  ngx_string("/data.mdb"),
+                                  ngx_string("/lock.mdb"),
+                                  ngx_null_string,
+                                };
+
+    ccf = (ngx_core_conf_t *) ngx_get_conf(cycle->conf_ctx, ngx_core_module);
+
+    if (ccf->user == (ngx_uid_t) NGX_CONF_UNSET_UINT) {
+        return NGX_OK;
+    }
+
+    for (name = names; name->len; name++) {
+        ngx_snprintf(name_buf, NGX_MAX_PATH,
+                     "%V%V%Z", &lcf->env_path->name, name);
+
+        if (ngx_file_info(name_buf, &fi) == NGX_FILE_ERROR) {
+            ngx_log_error(NGX_LOG_CRIT, cycle->log, ngx_errno,
+                          ngx_file_info_n " \"%s\" failed", name_buf);
+            return NGX_ERROR;
+        }
+
+        if (fi.st_uid != ccf->user) {
+            if (chown((const char *) name_buf, ccf->user, -1) == -1) {
```

Comment:
I do not know why we choose `chown` in the Nginx process. I think the right choice is that we must make sure the right permission at the Nginx creat file time.

Reply:
The master process may have the root privilege, for safety reasons, we should change them to the proper permission.